crates/bevy_asset/src/server/info.rs

```diff
@@ -189,7 +189,11 @@ impl AssetInfos {
         loading_mode: HandleLoadingMode,
         meta_transform: Option<MetaTransform>,
     ) -> Result<(UntypedHandle, bool), GetOrCreateHandleInternalError> {
-        match self.path_to_id.entry(path.clone()) {
+        // TODO: Need to re-work this to actually handle the None case
+        let type_id_unwrapped =
+            type_id.ok_or(GetOrCreateHandleInternalError::HandleMissingButTypeIdNotSpecified)?;
```

Comment:
Looking at where this function gets used, there is only in `crates/bevy_asset/src/server/mod.rs` line 401 where there is a chance of it being `None`. There is a fairly large comment I didn't read which probably explains why someone would want to pass a `None` as `type_id` here.

Reply:
> This contains Some(UntypedHandle), if it was retrievable
> If it is None, that is because it was _not_ retrievable, due to
>
>    1. The handle was not already passed in for this path, meaning we can't just use that
>    2. The asset has not been loaded yet, meaning there is no existing Handle for it
>    3. The path has a label, meaning the AssetLoader's root asset type is not the path's asset type
>
> __In the None case, the only course of action is to wait for the asset to load so we can allocate the
> handle for that type.__

Since the `None` case for `type_id` can only occur when there is a label (`path.label().is_none().then(|| loader.asset_type_id())`), there's no way to infer the `TypeId` of the asset to load, so there's no way to generate a valid `Handle` to return back on `AssetInfos::get_or_create_path_handle_internal`. Based on how `unwrap_with_context` handles the `GetOrCreateHandleInternalError::HandleMissingButTypeIdNotSpecified` error case (no panic, just converts it to `None`), I think it's reasonable to early-return `Err` in `AssetInfos::get_or_create_path_handle_internal` when `type_id` is `None`. However, this probably needs a refactor at some point, since it doesn't feel right. To have a function always return `None` if a particular input is `None`. Feels like the input should be required at that point.

Reply:
I've updated the `TODO` comment to reflect this information, but I'm going to stop short of replacing it with a function that explicitly only takes `TypeId` just to limit scope.